src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_load_store_emitters.cpp

```diff
@@ -65,18 +61,163 @@ void jit_load_emitter::emit_isa(const std::vector<size_t> &in_idxs, const std::v
+    BReg dst_b = BReg(out_idxs[0]);
+    HReg dst_h = HReg(out_idxs[0]);
+    SReg dst_s = SReg(out_idxs[0]);
+
+    switch (load_num_) {
+        case 0:
+            break;
+        case 1:
+            h->ldr(dst_b, ptr(src, byte_offset_));
+            break;
+        case 2:
+            h->ldr(dst_h, ptr(src, byte_offset_));
+            break;
+        case 3:
+            h->ldr(dst_h, ptr(src, byte_offset_));
+            h->add_imm(prc, src, byte_offset_ + 2 * sizeof(int8_t), h->X_DEFAULT_ADDR);
+            h->ld1(dst.b[2], ptr(prc));
+            break;
+        case 4:
+            h->ldr(dst_s, ptr(src, byte_offset_));
+            break;
+        default:
+            OV_CPU_JIT_EMITTER_THROW("Unexpected number of elements to load.");
+    }
+}
+
+template <cpu_isa_t isa>
+void jit_load_emitter::emit_isa(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs) const {
+    bool is_supported_precision = one_of(src_prc_, ov::element::f32, ov::element::i32, ov::element::f16, ov::element::i8, ov::element::u8) &&
+                                  (src_prc_ == dst_prc_ || one_of(dst_prc_, ov::element::f32, ov::element::i32));
+    OV_CPU_JIT_EMITTER_ASSERT(is_supported_precision, "Unsupported precision pair.");
+    OV_CPU_JIT_EMITTER_ASSERT(load_num_ <= static_cast<int>((get_vec_length() / dst_prc_.size())),
+                              "Unexpected number of elements to load.");
+
+    switch (src_prc_) {
+        case ov::element::f32:
+            load_qbyte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    break;
+                case ov::element::i32:
+                    cvt_f32_to_i32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                default:
+                    OV_CPU_JIT_EMITTER_THROW("Unsupported output type: ", dst_prc_.get_type_name());
+            }
+            break;
+        case ov::element::i32:
+            load_qbyte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    cvt_i32_to_f32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                case ov::element::i32:
+                    break;
+                default:
+                    OV_CPU_JIT_EMITTER_THROW("Unsupported output type: ", dst_prc_.get_type_name());
+            }
+            break;
+        case ov::element::f16:
+            load_dbyte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    cvt_f16_to_f32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                case ov::element::i32:
+                    cvt_f16_to_f32<isa>(h, aux_vec_idxs, aux_vec_idxs);
+                    cvt_f32_to_i32<isa>(h, aux_vec_idxs, out_idxs);
```

Comment:
One of the my main concerns about these emitters the following:
at the moment, `jit_load_emitter` and `jit_store_emitters` for different precisions looks like `Load + Convert` or `Convert + Store`. Then I don't see sense in the passes `FuseLoadConvert` and `FuseStoreConvert` 🤔 

On x64 we just use the special instructions to load 8 packed 8-bit integers to 8 packed 16-bit integers. Because of that the expression `LoadConvert` is more efficient than the pair `Load` + `Convert`.
Does aarch64 something the same?
If it does not, maybe there is no sense to support `Load` `any precision -> f32/i32` and `Store` `i32/f32 -> any precision`? If developers needs this conversion, just call convert emitter 🤔 
And no need to register these optimization passes in data flow pipeline in snippets which fuse `Convert` with memory expression.
By another hand, we can just call `convert_emitter` in `load/store_emitter` for precision conversion to support generic case. Then users don't need to think about different precisions of GPR and VEC when they will write own implementations (not snippets).
But I'd prefer first variant (as RISC-ideology😄  )
What do you think? It's open question for discussion

Reply:
Thanks Alexandra for the comment! I agree that from the perspective of RISC-ideology the first variant is better. Yet I vote for your second variant, for  the exact reason you provided. Applying the second variant, we align with the same behavior of x64 when call load/store_emitter. As load/store_emitter will be called in more places in non-snippetes JIT implementations, and for most of the cases they load from source precision to f32 (or store from f32 to dst precision), I believe such end to end second variant will bring more convinience for developers. Please feel free to have further discussions.

Reply:
@xuchen-intel, emitters for all architectures should be kept as simple as possible. We should not try to align semantics of load/store emitters between x86 and arm. 
x86 load/store emitters support intrinsically conversions only because there are hardware instructions that can simultaneously convert&write to memory. If there are no such instructions on arm, we should not try to incorporate convert emitters into load/store emitters. 
So my suggestion is to keep load/sore without convert functionality on arm. We just should not call the `FuseLoadStoreConvert` pass on arm, and appropriate convert emitters would be used automatically. This would allow us to simplify the load/store emitters.

Reply:
I've thought twice. I think both of you have made good points! And my concern about convinience of usage in non-snippets implementations can be solved by packing load/store and conversion emitters locally. So let's do this separation to make load/store as simple as possible!

For now, I've tried to remove the conversion emitter from load/store, but there show up many test case failures. I need to investigate the issue. So I created a separate ticket [150430](https://jira.devtools.intel.com/browse/CVS-150430) (cc' @dmitry-gorokhov ) to track this task. Thanks @IvanNovoselov  and @a-sidorova!